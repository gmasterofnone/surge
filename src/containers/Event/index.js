import React from 'react';
import { connect } from 'react-redux';
import { randomNumber } from '../../utils/Helper';
import { NavLink } from 'react-router-dom'
import { toggleFavorite } from '../../actions/index';

import addTrue from '../../assets/add-true.svg'
import addFalse from '../../assets/add-false.svg'
import share from '../../assets/share.svg'



import './Event.css'

export const EventCard = ( { event, toggleFavorite } ) => {
  const { title, image, body, comments, surge, attending, id, favorite } = event;
  let uuidv4 = require("uuid/v4");

 

  let randomStyle = randomNumber(0, 20000)
  
  let styleSheet = document.styleSheets[0];
  let keyframes = 
    `@keyframes surge-${randomStyle} {
      0% {width: 0%;}
      100% {width: ${surge}%;}
    }`

  let surgeStyle = 
    `#surge-${randomStyle} {
      -webkit-animation-name: surge-${randomStyle};
      animation-name: surge-${randomStyle};
      -webkit-animation-delay: .7s;
      animation-delay: .7s;
    }`
  
  styleSheet.insertRule(keyframes, styleSheet.cssRules.length);
  styleSheet.insertRule(surgeStyle, styleSheet.cssRules.length);

  const avatars = comments.map(comment => (
    <img className={`avatars`} 
      src={comment.avatar} 
      alt={comment.name} 
      key={uuidv4()} 
    />
  ))

  return(
    <div className='event'>
        <div className='event-image'
          style={{backgroundImage: `url(${image})`}}
        >
          {
            favorite &&
          <div className='event-favorite-container'
            style={{opacity: `1`}}
          >
            <img className='fav-btn' 
              src={addTrue} 
              alt='favorite button'
              onClick={() => toggleFavorite(event)}
            />
          </div>
          }
          {
            !favorite &&
          <div className='event-favorite-container'>
            <img className='fav-btn' 
              src={favorite ? addTrue : addFalse} 
              alt='favorite button'
              onClick={() => toggleFavorite(event)}
            />
            <img className='share-btn' src={share} alt='share button'/>
          </div>
          }
        </div>
        <NavLink className='event-link' to={`/${id}`}>
          <div className='event-info'>
            <div>
              <h3 className='event-title'>{title[0]}</h3>
              {/* <p className='event-body'>{body.slice(0, 130)}...</p> */}
            </div>
            <div className='event-interaction'>
              <ul className="surge-container-event">
                <label>{`Surge | ${attending} Followers`}</label>
                <li>
                  <span className="progressbar-event progressblue-event" id={`surge-${randomStyle}`}></span>
                </li>
              </ul>
              <div className='event-comment-avatars'>
                { avatars.slice(0, randomNumber(3, 5)) }
                <p className='event-comment-count'>{`${avatars.length} comments`}</p>
              </div>
            </div>
          </div>
      </NavLink>
      </div>
  )
}

export const mapDispatchToProps = (dispatch) => ({
  toggleFavorite: (article) => dispatch(toggleFavorite(article))
})

export default connect(null, mapDispatchToProps)(EventCard)