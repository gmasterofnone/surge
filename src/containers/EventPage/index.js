import React, { Component } from 'react';
import { connect } from 'react-redux'
import { randomNumber } from '../../utils/Helper';
import { Link } from 'react-router-dom'

import { toggleFavorite } from '../../actions/index';

import addTrue from '../../assets/add-true.svg'
import addFalse from '../../assets/add-false.svg'
import share from '../../assets/share.svg'
import { avatar } from '../../assets/'


 

import './EventPage.css'

export class EventPage extends Component {
  constructor() {
    super()
    this.state = {
      comments: []
    }
  }

  componentDidMount() {
    this.props.blur()
    this.loadComments()
  }

  loadComments = (userComment) => {
    const { comments } = this.props.event
    this.setState( { comments: [...this.state.comments, ...comments]} )
  }

  componentWillUnmount() {
    this.props.blur()
  }

  render() {
    const { title, image, body, surge, attending, favorite, source, date, link, author, event } = this.props.event;
    const { user } = this.props;
    console.log(user)
    let uuidv4 = require("uuid/v4");

    let randomStyle = randomNumber(0, 20000)
    
    let styleSheet = document.styleSheets[0];
    let keyframes = 
      `@keyframes surge-${randomStyle} {
        0% {width: 0%;}
        100% {width: ${surge}%;}
      }`

    let surgeStyle = 
      `#surge-${randomStyle} {
        -webkit-animation-name: surge-${randomStyle};
        animation-name: surge-${randomStyle};
        -webkit-animation-delay: .7s;
        animation-delay: .7s;
      }`
    
    styleSheet.insertRule(keyframes, styleSheet.cssRules.length);
    styleSheet.insertRule(surgeStyle, styleSheet.cssRules.length);

    const userComments = this.state.comments.map(comment => (
      <li className='user-comments' key={uuidv4()}>
        <img className='user-comments-image' src={comment.avatar} alt={comment.name}/>
        <div>
          <h4 className='event-user-name'>{comment.name}</h4>
          <p className='single-user-comment'>{comment.comment}</p>
        </div>
      </li>
    ))


    return(
      <div className='event-page-container'>
        <div className='event-page'>
          <p className='close-btn'><Link className='close-btn' to='/'>âœ•</Link></p>
          <div className='eventpage-image'
              style={{backgroundImage: `url(${image})`}}
          >
            <ul className="surge-container-event">
              <label className={surge ===100 ? 'surged' : ''}
              >{`${surge === 100 ? 'Surged' : `${surge}% Surged`} /// ${attending} Followers`}
              </label>
              <li>
                <span className="progressbar-event progressblue-event" id={`surge-${randomStyle}`}></span>
              </li>
            </ul>
          </div>
          <div>
            <p className='eventpage-source'>{`${source} - ${date}`}</p>
            <h1 className='eventpage-title'>{title[0]}</h1>
            <p className='eventpage-author'>{author}</p>
            <p className='eventpage-body'
            >{body}
              <a target='blank' href={link}>view original article.</a>
            </p>
          </div>   
          <p className='comments-tag'>Recent Comments</p>
          <ul className='user-comments-section'>
            {userComments}
          </ul>
          <form className='add-comment'>
            <div className='add-comment-photo'>
              <img className='user-comments-avatar' src={avatar[user.avatar]}/>
              <p className='add-photo'>add photo</p>
            </div>
            <div className='comment-input'>
              <input className='name-input'  
                placeholder='Enter name'
              />
              <input className='comment-input' 
                placeholder='Join the discussion and add a comment..'
              />
            </div>
          </form>
        </div>
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  user: state.user,
})

export const mapDispatchToProps = (dispatch) => ({
  toggleFavorite: (article) => dispatch(toggleFavorite(article))
})

export default connect(mapStateToProps, mapDispatchToProps)(EventPage)