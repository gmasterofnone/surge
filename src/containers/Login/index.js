import React from 'react';
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import { avatar } from '../../assets/'
import { createUser } from '../../actions/index';
import logo from '../../assets/logo.svg';


import './Login.css'

export const login = ( { createUser } ) => {
  let uuidv4 = require("uuid/v4");
  const avatars = Object.keys(avatar).map(icon => (
    <div className='avatar-container' key={uuidv4()}>
      <Link to='/'>
        <img 
          className='login-avatar'
          src={avatar[icon]} 
          alt='avatar' 
          value={icon}
          onClick={() => createUser(icon)}
          key={uuidv4()}
        />
      </Link>
      <img 
        className='login-logo'
        src={logo} 
        alt='logo' 
        value={icon}
        onClick={() => createUser(icon)}
        key={uuidv4()}
      />
    </div>
  ))

  return(
    <div className='login'>
      {/* <img className='avatar' 
        src={avatar.one} 
        alt='avatar' 
        value='one' 
      />
      <img className='avatar' 
        src={avatar.two} 
        alt='avatar' 
        value='one' 
      />
      <img className='avatar' 
        src={avatar.three} 
        alt='avatar' 
        value='one' 
      />
      <img className='avatar' 
        src={avatar.four} 
        alt='avatar' 
        value='one' 
      /> */}
      {avatars}
     
    </div>
  )

}

export const mapDispatchToProps = (dispatch) => ({
  createUser: (avatar) => dispatch(createUser(avatar))
})

export default connect(null, mapDispatchToProps)(login)