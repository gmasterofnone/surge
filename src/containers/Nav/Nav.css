header {
  display: flex;
  margin: 40px 0 0 50px;
  align-items: flex-end;
}

.add-margin {
  margin: 40px 0 40px 50px;
}

.avatar-logo {
  margin-right: 10px;
  position: relative;
}

.fav-count {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 22px;
  height: 22px;
  background-color: rgb(255, 255, 255, .9);
  border: 1px solid black;
  border-radius: 50%;
  position: absolute;
  top: 53px;
  left: 52px;
  cursor: pointer;
}

.fav-number {
  font-family: Arial, Helvetica, sans-serif;
  font-size: .75rem;
  font-weight: 600;
  margin: 0;
  padding-right: 1px;
}

.avatar {
  cursor: pointer;
  height: 80px;
  margin-right: 3px;
}

.logo {
  height: 70px;
  position: relative;
  top: 0px;
}

.search-btn {
  position: relative;
  height: 25px;
  margin-bottom: 20px;
  cursor: pointer;
  animation: bounce 1s;
}

.loading {
  margin: 0 8px 27px 0;
}

header img {
  transition: 2s;
}

@keyframes bounce {
  0% {
    margin-bottom: 20px;
  }
  30% {
    margin-bottom: 40px;
  }
  100% {
    margin-bottom: 20px;
  }
}

.topic-form {
  margin: 0 2px 24px 5px;
  height: 20px;
}

.search, .search-topics {
  font-size: 1.25rem;
  font-family: Arial, Helvetica, sans-serif;
  font-weight: 600;
}

.search {
  border: none;
  caret-color: #ff0022;
  font-weight: 600;
}

.search:focus {
  outline: none;
}

.search-topics {
  font-style: oblique;
  padding: 0 8px 0 0;
  margin-bottom: 20px;
}

.search-topics:hover {
  cursor: pointer;
  text-decoration: line-through;
  filter: contrast(.25)
}

.favorites-section {
  background-color: blue;
  height: 200px;
  width: 100%
}

