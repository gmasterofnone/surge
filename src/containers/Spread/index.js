import React from 'react';
import { connect } from 'react-redux'

import TopicContainer from '../../components/TopicContainer'

import './Spread.css'

export const Spread = ( { content } ) => {
  let uuidv4 = require("uuid/v4");
  const layouts = Object.keys(content).map(topic => (
    <TopicContainer key={uuidv4()} content={content[topic]} />
  ))

  return(
    <div className='Spread'>
      { layouts }
    </div>
  )
}

export const mapStateToProps = (state) => ({
  content: state.content
})


export default connect(mapStateToProps)(Spread)