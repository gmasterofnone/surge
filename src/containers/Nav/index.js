import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { getTopic } from '../../thunks/getTopic';
import { removeTopic, toggleFavorite, deleteUser } from '../../actions'
import {randomNumber } from '../../utils/Helper'

import logo from '../../assets/logo.svg';
import loading from '../../assets/loading.svg'
import add from '../../assets/add.svg';
import close from '../../assets/close.svg';

import { avatar } from '../../assets/'
import { color } from '../../assets/color'

import './Nav.css'


export class Nav extends Component {
  constructor() {
    super()
    this.state = {
      addTopic: false,
      search: '',
      showFavorites: false
    }
  }

  componentDidMount() {
    this.toggleAdd();
  }

  toggleAdd = () => {
    this.setState( {
      addTopic: !this.state.addTopic,
      search : ''
      } 
    );
  }

  showFavorites = () => {
    this.setState( { showFavorites: !this.state.showFavorites} ) 
  }

  handleChange = (event) => {
    const { value } =  event.target;
    this.setState( { search: value } );
  }

  addTopic = event => {
    event.preventDefault();
    const { search } = this.state;
    const randomColor = color[randomNumber(0, 4)];
    const topic = { search, randomColor };
    this.props.getTopic(topic)
    this.setState( { search: '', addTopic: !this.state.addTopic } );
  }

  removeTopic = (name) => {
   this.props.removeTopic(name)
  }

  closeAccount = () => {
    this.props.deleteUser();
    window.location.reload()
  }

  
  render() {
    const { user, isLoading, toggleFavorite } = this.props;
    const { search, addTopic, showFavorites } = this.state;
    let uuidv4 = require("uuid/v4");

    let displayTopics = [];
    let favoriteArticles = [];

    if (user.topics) {
      displayTopics = user.topics.map(topic => (
          <span className={`${topic} search-topics`}
            key={uuidv4()}
            style={{color: `${topic.randomColor}`}}
            onClick={() => this.removeTopic(topic.search)}
          >{topic.search.toUpperCase()},</span> 
      )) 
      favoriteArticles = user.favorites.map(favorite => (
        <li key={uuidv4()} className='fav-list'>
          <img className='fav-article-img' 
            src={favorite.image}
            alt='favorite article'
          />
          <div>
            <h3 
              className='fav-title'>{favorite.title[0].slice(0, 30).toUpperCase()}...
            </h3>
            <p 
              className='fav-delete'
              onClick={()=> toggleFavorite(favorite)}
            >delete
            </p>
          </div>
        </li>
      ))
    } 

    return(
      <div>
        <header>
          <div className='avatar-logo'>
            {
              user.favorites && user.favorites.length > 0
                ? <div className='fav-count' onClick={this.showFavorites}>
                    <p className='fav-number'>{user.favorites.length}</p>
                  </div>
                : null
            }
            <Link to='/'>
              <img 
                className='avatar'
                src={avatar[user.avatar]} 
                alt='avatar' 
                onClick={this.showFavorites}
              />
            </Link>
            <img 
              className='logo'
              src={logo} 
              alt='logo' 
            />
          </div>
          { displayTopics }
          {
            addTopic &&
            <form className='topic-form' 
              onSubmit={this.addTopic}
            >
              <input autoFocus className='search'
                value={search}
                placeholder='enter a topic'
                style={ search.length 
                  ? {width: `${(search.length * 10) + 15}px`}
                  : {width: `123px`}}
                onChange={this.handleChange}
              />
            </form>
          }
          {
            isLoading &&
            <img className='loading' src={loading} alt='loading'/>
          }
          <img 
            className='search-btn'
            src={addTopic ? close : add} 
            alt='add button'
            onClick={this.toggleAdd} 
          />
        </header>
        {
          showFavorites &&
          <div className='favorites-section'>
            <div>
              <div className='controls-container'>
                <div className='close-favs-btn' 
                  onClick={this.showFavorites}
                >
                  <p className='close-icon'>âœ•</p>
                </div>
                <p className='close-account'
                  onClick={this.closeAccount}
                >close account
                </p>
              </div>
            </div>
            <div>
              <ul className='favorites-container'>
                { favoriteArticles }
              </ul>
            </div>  
          </div>
        }
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  content: state.content,
  user: state.user,
  isLoading: state.isLoading
})

export const mapDispatchToProps = (dispatch) => ({
  getTopic: (topic) => dispatch(getTopic(topic)),
  removeTopic: (topic) => dispatch(removeTopic(topic)),
  toggleFavorite: (topic) => dispatch(toggleFavorite(topic)),
  deleteUser: () => dispatch(deleteUser())
})

export default connect(mapStateToProps, mapDispatchToProps)(Nav)