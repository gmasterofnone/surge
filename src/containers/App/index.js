import React, { Component } from 'react';
import { connect } from 'react-redux'
import { Route, withRouter, Switch } from 'react-router-dom';
import { loginUser } from '../../actions/index'
import { checkUser } from '../../utils/Helper'

import Nav from '../Nav'
import Login from '../Login';
import Spread from '../Spread';


import './App.css';


class App extends Component {
  
  async componentDidMount() {
    const user = checkUser()
    
    user 
      ? this.props.loginUser(user)
      : this.props.history.push('/login')
  }

  render() {

    return (
      <div className="App">
        <Switch>
          <Route exact path='/login' render={() => <Login />}/>
          <Route path='/' render={() => <Nav />}/>
        </Switch>
        <Spread />
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  isLoading: state.isLoading,
  hasErrored: state.hasErrored,
  content: state.content,
  user: state.user
})

export const mapDispatchToProps = (dispatch) => ({
  loginUser: (user) => dispatch(loginUser(user))
})

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App));
