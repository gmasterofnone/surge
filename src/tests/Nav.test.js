import React from 'react';
import { Nav } from '../containers/Nav';
import { mapStateToProps, mapDispatchToProps } from '../containers/Nav';
import { shallow } from 'enzyme';

describe('Nav', () => {
  let wrapper;
  let mockAddTopic;
  let mockRemoveTopic
  let mockUser;

  beforeEach(() => {
    mockUser = { topics: [], favorites: [{id: 1}]};
    mockAddTopic = jest.fn()
    mockRemoveTopic = jest.fn()
    wrapper = shallow(
      <Nav
        history={[]}
        user={mockUser}
        getTopic={mockAddTopic}
        removeTopic={mockRemoveTopic}
      />
    )
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should allow the user to add a topic', () => {
    wrapper.find('.search-btn').simulate('click')
    expect(wrapper.state().addTopic).toEqual(true);
  });

  it('should add a topic state', () => {
    wrapper.find('.search-btn').simulate('click')
    const event = {
      preventDefault: () => jest.fn,
      target: {
        value: 'hello'
      }
    }
    wrapper.instance().handleChange(event)

    expect(wrapper.state().search).toEqual('hello');
  });

  it('should add a topic to the store', () => {
    wrapper.find('.search-btn').simulate('click')
    const event = {
      preventDefault: () => jest.fn,
      target: {
        value: 'hello'
      }
    }
    wrapper.instance().addTopic(event)

    expect(wrapper.state().search).toEqual('');
    expect(wrapper.state().addTopic).toEqual(false);
  });

  it('should remove a topic to the store', () => {
    wrapper.find('.search-btn').simulate('click')
    const name = 'immigration'
    wrapper.instance().removeTopic(name)
  });

  it('should toggle favorites section', () => {
    wrapper.find('.avatar').simulate('click')

    expect(wrapper.state().showFavorites).toEqual(true);
  });

})

describe('mapStateToProps', () => {
  it('should have access to loading status and users', () => {
    const mockStore = {
      isLoading: false,
      user: {}
    }
    const expected = {...mockStore};
    const result = mapStateToProps(mockStore);
    expect(result).toEqual(expected);
  });
})

describe('mapDispatchToProps', () => {
  it('should call dispatch when getTopic is invoked', () => {
    const mockDispatch = jest.fn();
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.getTopic({search: 'immigration'});
    expect(mockDispatch).toHaveBeenCalled();
  });

  it('should call dispatch when removeTopic is invoked', () => {
    const mockDispatch = jest.fn();
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.removeTopic({});
    expect(mockDispatch).toHaveBeenCalled();
  });

})