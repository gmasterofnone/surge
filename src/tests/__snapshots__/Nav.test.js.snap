// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match the snapshot 1`] = `
<header>
  <div
    className="avatar-logo"
  >
    <Link
      replace={false}
      to="/"
    >
      <img
        alt="avatar"
        className="avatar"
      />
    </Link>
    <img
      alt="logo"
      className="logo"
      src="logo.svg"
    />
  </div>
  <img
    alt="add button"
    className="search-btn"
    onClick={[Function]}
    src="add.svg"
  />
</header>
`;
